sbi/utils/tensorboard_output.py

```diff
@@ -0,0 +1,213 @@
+    scalars = all_event_data["scalars"]
+
+    if not disable_tensorboard_prompt:
+        logger.warning(
+            (
+                "For an interactive, detailed view of the summary, launch tensorboard "
+                f" with 'tensorboard --logdir={log_dir}' from a"
+                " terminal on your machine, visit http://127.0.0.1:6006 afterwards."
+                " Requires port forwarding if tensorboard runs on a remote machine, as"
+                " e.g. https://stackoverflow.com/a/42445070/7770835 explains.\n"
+            )
+        )
+        logger.warning(f"Valid tags are: {sorted(list(scalars.keys()))}.")
+
+    _check_tags(scalars, tags)
+
+    if len(scalars[tags[0]]["step"]) == tensorboard_scalar_limit:
+        logger.warning(
+            (
+                "Event data as large as the chosen limit for tensorboard scalars."
+                "Tensorboard might be subsampling your data, as "
+                "https://stackoverflow.com/a/65564389/7770835 explains."
+                " Consider increasing tensorboard_scalar_limit to see all data.\n"
+            )
+        )
+
+    plot_options = _get_default_opts()
+
+    plot_options.update(figsize=figsize, fontsize=fontsize)
+    if fig is None or axes is None:
+        fig, axes = plt.subplots(
+            1, len(tags), figsize=plot_options["figsize"], **plot_options["subplots"]
+        )
+    axes = np.atleast_1d(axes)
+
+    ylabel = ylabel or tags
+
+    for i, ax in enumerate(axes):
+        ax.plot(scalars[tags[i]]["step"], scalars[tags[i]]["value"], **plot_kwargs)
+
+        ax.set_ylabel(ylabel[i], fontsize=fontsize)
+        ax.set_xlabel(xlabel, fontsize=fontsize)
+        ax.xaxis.set_tick_params(labelsize=fontsize)
+        ax.yaxis.set_tick_params(labelsize=fontsize)
+
+    plt.subplots_adjust(wspace=0.3)
+
+    return fig, axes
+
+
+def list_all_logs(inference: _NeuralInference) -> List:
+    """Returns a list of all log dirs for an inference object."""
+    method = inference.__class__.__name__
+    log_dir = Path(get_log_root()) / method
+    return sorted(log_dir.iterdir())
+
+
+def _get_event_data_from_log_dir(
+    log_dir: Union[str, Path], size_guidance=DEFAULT_SIZE_GUIDANCE
+) -> Dict[str, Dict[str, Dict[str, List[Any]]]]:
+    """Returns an exhaustive nested dictionary of all event data stored by tensorboards
+    SummaryWriter.
+
+    The event data is stripped off from their native tensorboard event types and
+    represented in a tabular
+    way, i.e. Dict[str, List].
+
+    The hierarchy of the dictionary is:
+        1. tag type: event types that can be logged with tensorboard like 'scalars',
+           'images', 'histograms', etc.
+        2. tag: tag for the event type that the user of the SummaryWriter specifies.
+        3. tag type attribute: attribute of the event.
+
+    Args:
+        log_dir
+        size_guidance: to avoid OOming. Defaults to tensorboards default size_guidance.
```

Comment:
What's OOming?

Reply:
OOM = out of memory, because it loads most event data to memory at once